cernopendata_client/cli.py

```diff
@@ -152,7 +158,25 @@ def get_file_locations(server, recid, doi, title, protocol, expand):
     type=click.STRING,
     help="Which CERN Open Data server to query? [default=http://opendata.cern.ch]",
 )
-def download_files(server, recid, doi, title, protocol, expand):
+@click.option(
+    "--filter-name",
+    "name",
+    type=click.STRING,
+    help="Download file that contains the specified string",
```

Comment:
If we look for matching *inside* file name, then the option `--filter-name` plays almost the same role as `--filter-regexp`, and we would be loosing important functionality to match "exactly".  So I suggest to change the meaning of `--filter-name` to make *exact match* for the full filename.  In this way it makes sense to have different options for file name matching. (Although in theory one could do with a single regexp matching option when user would write `^myfilename\.pdf$` but that would be less readable...)

- filter-name ... change to exact matching
- filter-regexp ... good, keep as is
- filter-range -- good, keep as is

When you

Reply:
The exact filter is not working, here is the use case:

```console
$ cernopendata-client download-files --recid 5500 --filter-name nothing
==> Downloading file 1 of 11
==> Downloading file: ./5500/BuildFile.xml
==> Downloading file 2 of 11
==> Downloading file: ./5500/HiggsDemoAnalyzer.cc
==> Downloading file 3 of 11)
==> Downloading file: ./5500/List_indexfile.txt
==> Downloading file 4 of 11
==> Downloading file: ./5500/M4Lnormdatall.cc
==> Downloading file 5 of 11)
==> Downloading file: ./5500/M4Lnormdatall_lvl3.cc
==> Downloading file 6 of 11)
==> Downloading file: ./5500/demoanalyzer_cfg_level3MC.py
==> Downloading file 7 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level3data.py
==> Downloading file 8 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level4MC.py
==> Downloading file 9 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level4data.py
==> Downloading file 10 of 11
==> Downloading file: ./5500/mass4l_combine.pdf
==> Downloading file 11 of 11
==> Downloading file: ./5500/mass4l_combine.png
Downloading: 90/90 kiB (100%)
Download completed!
``` 

I.e. when no match is found, it downloads everything... it should simply do nothing in such a case.

Reply:
Ditto for regexp filter:

```console
$ $ cernopendata-client download-files --recid 5500 --filter-regexp Higgs
==> Downloading file 1 of 11
==> Downloading file: ./5500/BuildFile.xml
==> Downloading file 2 of 11
==> Downloading file: ./5500/HiggsDemoAnalyzer.cc
==> Downloading file 3 of 11)
==> Downloading file: ./5500/List_indexfile.txt
==> Downloading file 4 of 11
==> Downloading file: ./5500/M4Lnormdatall.cc
==> Downloading file 5 of 11)
==> Downloading file: ./5500/M4Lnormdatall_lvl3.cc
==> Downloading file 6 of 11)
==> Downloading file: ./5500/demoanalyzer_cfg_level3MC.py
==> Downloading file 7 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level3data.py
==> Downloading file 8 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level4MC.py
==> Downloading file 9 of 11
==> Downloading file: ./5500/demoanalyzer_cfg_level4data.py
==> Downloading file 10 of 11
==> Downloading file: ./5500/mass4l_combine.pdf
==> Downloading file 11 of 11
==> Downloading file: ./5500/mass4l_combine.png
Downloading: 90/90 kiB (100%)
Download completed!
```

Should download only some files...

Reply:
Added this edge case when there are filters applied and no files match with the filters, no files will be downloaded